src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1533,6 +1535,36 @@ export class ResultsViewPageStore {
         invoke: () => this.getClinicalData("SAMPLE", this.studies.result!, this.samples.result, ["CANCER_TYPE", "CANCER_TYPE_DETAILED"])
     }, []);
 
+    readonly facetsClinicalDataForSamples = remoteData({
+        await: () => [
+            this.studies,
+            this.samples
+        ],
+        invoke: () => this.getClinicalData("SAMPLE", this.studies.result!, this.samples.result, ["FACETS_WGD", "FACETS_PURITY"])
+    }, []);
+    
+    @computed get sampleIds(): string[]
+    {
+        if (this.samples.result) {
+            return this.samples.result.map(sample => sample.sampleId);
+        }
+
+        return [];
+    }
+
+    readonly facetsClinicalDataGroupedBySample = remoteData({
+        await: () => [this.facetsClinicalDataForSamples],
+        invoke: async() => groupBySampleId(this.sampleIds, this.facetsClinicalDataForSamples.result)
+    }, []);
+
+    readonly clinicalDataGroupedBySampleMap = remoteData({
+        await: () => [this.facetsClinicalDataGroupedBySample],
+        invoke: async() => {
```

Comment:
just do this to minimize redundancies (no need for a local variable `sampleIdToClinicalDataMap` or the `return` statement):
```suggestion
        invoke: async() => mapSampleIdToClinicalData(this.facetsClinicalDataGroupedBySample.result, 'id', 'clinicalData')
```